shared/lib_battery_dispatch_pvsmoothing_fom.h

```diff
@@ -106,35 +106,22 @@ class dispatch_pvsmoothing_front_of_meter_t : public dispatch_automatic_t
 	/// Return intermediate calculations for validation - unscale by nameplate = min(system ac rating, grid interconnection limit)
     double batt_dispatch_pvs_outpower() { return  m_batt_dispatch_pvs_nameplate_ac * m_batt_dispatch_pvs_outpower; };
     double batt_dispatch_pvs_battpower() { return   m_batt_dispatch_pvs_nameplate_ac * m_batt_dispatch_pvs_battpower; };
-    double batt_dispatch_pvs_battsoc() { return   m_batt_dispatch_pvs_nameplate_ac * m_batt_dispatch_pvs_battsoc ; };
+    double batt_dispatch_pvs_battsoc() { return     m_batt_dispatch_pvs_nameplate_ac * m_batt_dispatch_pvs_battsoc ; };
```

Comment:
I think some of our output woes are coming from this line. It doesn't make sense that an output with units of % would get multiplied by a value in kW. Apologies for not catching this before.

Reply:
Scaling updated but now the woes are coming from the resting SOC input for the pv smoothing not affecting the algorithm.
Good agreement within the ramp interval with the SOC values:
<img width="1915" alt="image" src="https://user-images.githubusercontent.com/6498311/154481436-c515981e-2e8a-4896-9f8f-818fdcf975b3.png">

Issues with the low resting SOC  ( about 7% instead of the entered 50% for case shown ) causing the battery not to be able to discharge at critical ramp times and causing anti smoothing effects:
<img width="1911" alt="image" src="https://user-images.githubusercontent.com/6498311/154482002-4ca59871-e966-4cb6-9340-f895538e36de.png">

There seems to be an issue with the SOC value used in the pv smoothing dispatch calculated in the charge controller which uses the dispatch model as input...maybe we can average the SOC input over the ramp interval instead of taking the instantaneous value - which does not correspond exactly in the data tables, e.g. 
<img width="536" alt="image" src="https://user-images.githubusercontent.com/6498311/154483606-8d0f4d54-2e4b-4d86-ab42-9fd2fca234a0.png">


The tests have not been updated because the resting soc issue has not been resolved.

Reply:
The issue here was pv smoothing SOC was reporting smoothing + the next timestep's worth of power. With the changes in cefbcf0a865c377040ed3fbc5b8f12a1f1829316 - the SOC values should line up at the step when the smoothing dispatch was calculated, and then the battery SOC should increase or decrease until the next dispatch step. The second chart in https://github.com/NREL/ssc/pull/758#discussion_r809473365 shows this behavior.

Reply:
Some places where the smoothed power is a little suspect:
<img width="1889" alt="image" src="https://user-images.githubusercontent.com/6498311/154581110-aaa94222-bc45-440d-b595-37662498c01d.png">

Overall, very good agreement and much better - all tests passing on windows.

Reinstating all tests in last commit and the will merge after Github actions pass...